modules/OAuth2AuthCodes/views/view.authorize.php

```diff
@@ -0,0 +1,148 @@
+<?php
+/**
+*
+* SugarCRM Community Edition is a customer relationship management program developed by
+* SugarCRM, Inc. Copyright (C) 2004-2013 SugarCRM Inc.
+*
+* SuiteCRM is an extension to SugarCRM Community Edition developed by SalesAgility Ltd.
+* Copyright (C) 2011 - 2019 SalesAgility Ltd.
+*
+* This program is free software; you can redistribute it and/or modify it under
+* the terms of the GNU Affero General Public License version 3 as published by the
+* Free Software Foundation with the addition of the following permission added
+* to Section 15 as permitted in Section 7(a): FOR ANY PART OF THE COVERED WORK
+* IN WHICH THE COPYRIGHT IS OWNED BY SUGARCRM, SUGARCRM DISCLAIMS THE WARRANTY
+* OF NON INFRINGEMENT OF THIRD PARTY RIGHTS.
+*
+* This program is distributed in the hope that it will be useful, but WITHOUT
+* ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+* FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for more
+* details.
+*
+* You should have received a copy of the GNU Affero General Public License along with
+* this program; if not, see http://www.gnu.org/licenses or write to the Free
+* Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+* 02110-1301 USA.
+*
+* You can contact SugarCRM, Inc. headquarters at 10050 North Wolfe Road,
+* SW2-130, Cupertino, CA 95014, USA. or at email address contact@sugarcrm.com.
+*
+* The interactive user interfaces in modified source and object code versions
+* of this program must display Appropriate Legal Notices, as required under
+* Section 5 of the GNU Affero General Public License version 3.
+*
+* In accordance with Section 7(b) of the GNU Affero General Public License version 3,
+* these Appropriate Legal Notices must retain the display of the "Powered by
+* SugarCRM" logo and "Supercharged by SuiteCRM" logo. If the display of the logos is not
+* reasonably feasible for technical reasons, the Appropriate Legal Notices must
+* display the words "Powered by SugarCRM" and "Supercharged by SuiteCRM".
+*/
+
+if (!defined('sugarEntry') || !sugarEntry) {
+    die('Not A Valid Entry Point');
+}
+
+use League\OAuth2\Server\AuthorizationServer;
+use Api\V8\OAuth2\Entity\UserEntity;
+
+/**
+* Class Oauth2AuthCodesViewAuthorize
+*/
+class Oauth2AuthCodesViewAuthorize extends SugarView
+{
+    /**
+    * @var array $options
+    * Options for what UI elements to hide/show/
+    */
+    public $options = array(
+        'show_header' => false,
+        'show_title' => false,
+        'show_subpanels' => false,
+        'show_search' => false,
+        'show_footer' => false,
+        'show_javascript' => true,
+        'view_print' => false,
+    );
+
+    public function display()
```

Comment:
Not required but it would be good to reduce the cognitive complexity of this function.

Reply:
That is absolutely true. I thought about moving the decision points (show the authorization page or redirect once the decision has been made) to the controller class. However, we need the \Slim\App and \AuthorizationServer instances to validate the request, show the decision form, and to send the response. Since I cannot send the $app and $server instances to the display() function as params (the display function is called outside the module), there is no way to use the same AuthorizationServer instance in the controller and the view classes. I did not want to create those instances twice and thus ended up doing everything in the display function. 
There are two options that might work:
1. Attach the \Slim\App and \AuthorizationServer instances to an instance of \OAuth2AuthCodes, which is created at the beginning of the display() function. We could then use multiple functions of the \OAuth2AuthCodes class to do the logic. 
2. Split the display function into several function within the Oauth2AuthCodesViewAuthorize class. That might be easier to read, but still has the logic within the view class, where is should not be...

@Dillon-Brown: Do you have a preferred option? Or shall I leave the complex display function as it is for now?